### ISSUE TITLE
Release process: use github actions and release to bazel central registry

### ISSUE THREAD

**Issue by alexeagle** [GoogleContainerTools/container-structure-test#345]
Discussed with @loosebazooka as part of adding a Bazel rule in this repo: we'd like the release process to:
- build the binaries for all platforms on GitHub Actions so it can be fully community-owned without a Google Cloud project required to use Cloud Build
- the Bazel rule wants to know the checksums of the binaries, currently I just mirrored the ones of the previous release in https://github.com/GoogleContainerTools/container-structure-test/blob/main/repositories.bzl#L5-L14
- the release should include an artifact which is a zip of this repo, so that Bazel users can fetch the rule without depending on GH to serve an artifact (might be forced to do this for the Bazel Central Registry, based on the outcome in https://github.com/bazelbuild/bazel-central-registry/pull/607)
- it should automagically publish to the Bazel Central Registry on each release using https://github.com/bazel-contrib/publish-to-bcr

**Comment by alexeagle** [comment]
Note, I discussed this a bit with @gregmagolan since this is exactly the same shape as the copy_to_directory Go toolchain we build and publish with https://github.com/aspect-build/bazel-lib - we could probably share the approach.

**Comment by darkrift** [comment]
@loosebazooka Any chance you can take a look at this, specifically for the BCR release ?

**Comment by loosebazooka** [comment]
I think maybe @alexeagle or @thesayyn might have to advise. My bazel expertise is lacking. We could remove the goreleaser stuff and just do the build with bazel and I think we might be good?

**Comment by thesayyn** [comment]
Yes, we need to Bazelify the build process first, and do what Alex is suggesting here. 

> Note, I discussed this a bit with @gregmagolan since this is exactly the same shape as the copy_to_directory Go toolchain we build and publish with https://github.com/aspect-build/bazel-lib - we could probably share the approach.

**Comment by MPV** [comment]
I've seen releases since last update in this issue, is the above plan still accurate?
